agent/local/state_test.go

```diff
@@ -2195,3 +2196,48 @@ func drainCh(ch chan struct{}) {
 		}
 	}
 }
+
+func TestState_SyncChanges_DuplicateAddServiceOnlySyncsOnce(t *testing.T) {
+	state := local.NewState(local.Config{}, hclog.New(nil), new(token.Store))
+	rpc := &fakeRPC{}
+	state.Delegate = rpc
+	state.TriggerSyncChanges = func() {}
+
+	srv := &structs.NodeService{
+		Kind:           structs.ServiceKindTypical,
+		ID:             "the-service-id",
+		Service:        "web",
+		EnterpriseMeta: *structs.DefaultEnterpriseMeta(),
+	}
+	tok := "the-token"
+	err := state.AddServiceWithChecks(srv, nil, tok)
```

Comment:
If there are checks does this still work? A lot of registrations do have checks and I think those are compared separately so would still cause RPC as the checks are not being compared/set as in sync.

Reply:
Ah yes, that is correct. I've fixed the commit to do the same operation on checks as well.